### ISSUE TITLE
Html error page causing issues for GRPC

### ISSUE THREAD

**Issue by joekaus** [nghttp2/nghttp2#577]
We are using nghttpx to load balance http/2 traffic to grpc servers.
If one of the grpc servers is shutdown, an html error page is sent back to
the client. This causes an error in the client because it does not know how to parse
html.

We would like to be able to disable the generation of the html error page including the 
headers and instead notify the client of an error another way, perhaps using rst_stream.

**Comment by tatsuhiro-t** [comment]
The recent nghttpx has --error-page option to set the content of error page.  Is gRPC still not happy with empty error page?

**Comment by tatsuhiro-t** [comment]
FYI, yesterday I committed a change to retry request to the other backend when nghttpx cannot connect to  one backend.  It may reduce the exposure of 5xx page.

**Comment by joekaus** [comment]
Even with an empty error page, grpc returns an error. This is because the content-type is text/html and it refuses to accept it. However, it seems that your most recent change fixes the underlying issue, when the backend server goes down it is able to use another backend without error. Thank you for looking into this.

**Comment by tatsuhiro-t** [comment]
Does RST_STREAM cause different kind of error or graceful error message in gRPC client compared to the 503 response?  They are both errors, so the final outcome is probably the same.

**Comment by joekaus** [comment]
The 503 response gives ```code = 9 desc = transport: received the unexpected content-type "text/html; charset=UTF-8"```. Using RST_STREAM would give another error message depending on the nature of the error. But yes, they are both errors, so the client can look for this kind of error and process it accordingly.

**Comment by tatsuhiro-t** [comment]
Thanks.  So if I understand this correctly, this is nice-to-have feature, and not show stopper.

**Comment by joekaus** [comment]
Yes. The changes you made yesterday were actually what we needed. This would have just helped give another error message that we could handle on the client side. Thanks again.
